src/biome/text/data/datasource.py

```diff
@@ -191,14 +186,12 @@ def from_yaml(
             cfg_dict = yaml.safe_load(yaml_file)
 
         # File system paths are usually specified relative to the yaml config file -> they have to be modified
-        # path_keys is not necessary, but specifying the dict keys
+        # 'path_keys' is not necessary, but specifying the dict keys
         # (for which we check for relative paths) is a safer choice
-        path_keys = ["path", "source"]
-        make_paths_relative(os.path.dirname(file_path), cfg_dict, path_keys=path_keys)
+        make_paths_relative(os.path.dirname(file_path), cfg_dict, path_keys=["source"])
 
         if not cfg_dict.get("mapping"):
             cfg_dict["mapping"] = default_mapping or {}
-        cfg_dict.update(cfg_dict.pop("attributes", {}))
```

Comment:
Personally i like the config file to reflect the init of the class, it gives you a quick way to lookup the required format of the config file.

So if we want to keep the `attribute` key in the config, i would also like to keep it in the init ... @frascuchon would that be ok for you?

Reply:
Could we just keep the original `Datasource` config and think about a better `attribute` naming?

We can do a better experience analysis of creating `DataSource`s after this PR since we need to pass them to pipeline methods

Reply:
@frascuchon I kept the behavior of accepting `**kwargs` in the init, but grouping them together as `reader_options` in the yaml config.
@dvsrepo @frascuchon Are you ok with the name `reader_options`?